### PR TITLE
Fix #3300 LeakageWithCoefficient

### LINKED ISSUE #3300: LeakageWithCoefficient should have a port for reluctance input
The model Modelica.Magnetic.FluxTubes.Basic.LeakageWithCoefficient has an input signal, R_mUsefulTot, that provides the total reluctance of the path in which it is connected. To make this part usable in a GUI, that input should be a port to which a real signal can be connected. 

Actually, it's not clear to me why this quantity is an input and not a parameter. Are there examples of a model where it varies?  The doc-string for the component ("... not for dynamic simulation of actuators") suggests it isn't intended to be used in that application.

### PR DESCRIPTION
**Pull request by AHaumer**
fixes #3300

### CODE REVIEW

File: Modelica/Magnetic/FluxTubes/Basic/LeakageWithCoefficient.mo [hunk 1]
```diff
@@ -1,26 +1,26 @@
 within Modelica.Magnetic.FluxTubes.Basic;
 model LeakageWithCoefficient
   "Leakage reluctance with respect to the reluctance of a useful flux path (not for dynamic simulation of actuators)"
-
   extends BaseClasses.Leakage;
-
-  parameter SI.CouplingCoefficient c_usefulFlux=0.7
+  import Modelica.Constants.eps;
+  parameter SI.CouplingCoefficient c_usefulFlux(final min=eps, final max=1-eps)
     "Ratio useful flux/(leakage flux + useful flux) = useful flux/total flux";
```
Comment (beutlich) [inline Modelica/Magnetic/FluxTubes/Basic/LeakageWithCoefficient.mo]: I believe it is not a good idea to remove the default value here.

File: Modelica/Magnetic/FluxTubes/Examples/BasicExamples/QuadraticCoreAirgap.mo [hunk 2]
```diff
@@ -57,59 +57,75 @@ parameter Integer N=500 "Number of turns of exciting coil";
...
   connect(electricGround2.p, voltageSensor.n)
-    annotation (Line(points={{50,-10},{60,-10}}, color={0,0,255}));
+    annotation (Line(points={{60,-10},{70,-10}}, color={0,0,255}));
   connect(excitingCoil.port_n, magFluxSensor.port_n)
-    annotation (Line(points={{-20,-10},{-20,-20}}, color={255,127,0}));
+    annotation (Line(points={{-30,-10},{-30,-20}}, color={255,127,0}));
   connect(magFluxSensor.port_p, magneticGround.port)
-    annotation (Line(points={{-20,-40},{-20,-50}}, color={255,127,0}));
+    annotation (Line(points={{-30,-40},{-30,-50}}, color={255,127,0}));
+  connect(leakage.port_n, airGap.port_n) annotation (Line(points={{-1.77636e-15,
```
Comment (beutlich) [inline Modelica/Magnetic/FluxTubes/Examples/BasicExamples/QuadraticCoreAirgap.mo]:
```suggestion
  connect(leakage.port_n, airGap.port_n) annotation (Line(points={{0,
```

File: Modelica/Magnetic/FluxTubes/Examples/BasicExamples/QuadraticCoreAirgap.mo [hunk 3]
```diff
@@ -57,59 +57,75 @@ parameter Integer N=500 "Number of turns of exciting coil";
...
+          -10},{0,-10},{0,-40},{30,-40}}, color={255,127,0}));
+  connect(measuringCoil.port_p, leakage.port_p)
+    annotation (Line(points={{30,10},{1.77636e-15,10}}, color={255,127,0}));
```
Comment (beutlich) [inline Modelica/Magnetic/FluxTubes/Examples/BasicExamples/QuadraticCoreAirgap.mo]:
```suggestion
    annotation (Line(points={{30,10},{0,10}}, color={255,127,0}));
```

**Comment by AHaumer** [comment]
@christiankral 
In fact the component only models the leakage, not the main path - but it depends on the main path. 
I'm not sure what was the intention of Thomas Bödrich - this component has not been used in the examples ... I could guess: model a leakage defined by a fixed stray factor dependent on a variable main reluctance.

**Comment by JoeRiel** [comment]
While I'm happy with the change, it's still not clear how this would be usefully employed. An example would be helpful. How would the varying reluctance of the main path be modeled?

**Comment by AHaumer** [comment]
@JoeRiel 
Simply with a shape with nonlinear material characteristic

**Comment by JoeRiel** [comment]
Maybe I'm missing something obvious, but how does one generate the reluctance, so it can be passed to a LeakageWithCoefficient block, for a shape with a nonlinear magnetic characteristic?

**Comment by christiankral** [comment]
> @christiankral
> In fact the component only models the leakage, not the main path - but it depends on the main path.
> I'm not sure what was the intention of @ThomasBoedrich - this component has not been used in the examples ... I could guess: model a leakage defined by a fixed stray factor dependent on a variable main reluctance.

Yes, you're right. It is just a model of the leakage path.

**Comment by AHaumer** [comment]
@JoeRiel 
The shapes calculate the reluctance resp. permeance dependent on the actual point of operation. Have a look at the Example `BasicExamples.QuadraticCoreAirgap` of my PR.

**Comment by JoeRiel** [comment]
Thanks, Anton, for that example. That also suggests why R_usefulTotal was originally not designed as a port; it has to come from an internal signal.

**Comment by christiankral** [comment]
It makes sense to include variable reluctance and permeance models, too, if we provide a signal input for `LeakageWithCoefficient`: see #3311

**Comment by AHaumer** [comment]
@JoeRiel 
One could provide a signal output at the shapes for that purpose. Looks a little bit ugly ...

**Comment by christiankral** [comment]
I guess we are ready to merge now.

**Pull request merged**